flutter-for-react-native.md

```diff
@@ -0,0 +1,2007 @@
+
+<!-- skip -->
+{% prettify javascript %}
+// JavaScript
+_getIPAddress = () => {
+  const url="https://httpbin.org/ip";
+  return fetch(url)
+    .then(response => response.json())
+    .then(responseJson => {
+      console.log(responseJson.origin);
+    })
+    .catch(error => {
+      console.error(error);
+    });
+};
+{% endprettify %}
+
+Whereas Dart uses [`Future`](https://www.dartlang.org/tutorials/language/futures) objects to handle this.
+
+<!-- skip -->
+{% prettify dart %}
+// Dart
+_getIPAddress() {
+  final url = 'https://httpbin.org/ip';
+  Future<HttpRequest> request = HttpRequest.request(url);
+  request.then((value) {
+      print(JSON.decode(value.responseText)['origin']);
+  }).catchError((error) => print(error));
+}
+{% endprettify %}
+
+Try it out in [DartPad](https://dartpad.dartlang.org/b68eb981456c5eec03daa3c05ee59486).
+
+**Note:** Check [here](https://www.dartlang.org/tutorials/language/futures) for more details on future.
+
+#### async / await
+
+The async function declaration defines an asynchronous function. In JavaScript, when an async function is called, it returns a `Promise`. The `await` operator is used to wait for a `Promise`.
+
+<!-- skip -->
+{% prettify javascript %}
+// JavaScript
+async _getIPAddress() {
+  const url="https://httpbin.org/ip";
+  const response = await fetch(url);
+  const json = await response.json();
+  const data = await json.origin;
+  console.log(data);
+}
+{% endprettify %}
+
+Whereas in Dart, an `async` function returns a `Future`, and the body of the function is scheduled for execution later. The `await` operator is used to wait for a `Future`.
+
+<!-- skip -->
+{% prettify dart %}
+// Dart
+_getIPAddress() async {
+  final url = 'https://httpbin.org/ip';
+  var request = await HttpRequest.request(url);
+  String ip = JSON.decode(request.responseText)['origin'];
+  print(ip);
+}
+{% endprettify %}
+
+Try it out in [DartPad](https://dartpad.dartlang.org/96e845a844d8f8d91c6f5b826ef38951).
+
+**Note:** Check [here](https://www.dartlang.org/articles/language/await-async) for more details on async/await.
+
+#### Streams
+
+A Stream is a sequence of ongoing events that are ordered in time. A Stream is similar to a Future, but a Stream is a sequence of asynchronous events, whereas a Future represents a means for getting a single value sometime in the future.
+
+A Stream can emit three different things:
+* a value (of some type)
+* an error
+* a "completed" signal.
+  
+Streams are cheap and ubiquitous. Anything can be a stream: variables, user inputs, properties, caches, data structures, etc.
+Generator functions afford us the power of streams in JavaScript. 
+
+Consider this example in Javascript that makes use of the Generator function:
+
+<!-- skip -->
+{% prettify javascript %}
+// JavaScript
+var sum=0;
+function* sumStream () {
+  var n = 1;
+  while (true) {
+    yield sum=sum+n++;
```

Comment:
this is more equivalent to the sync*/Iterable pattern than the async*/Stream pattern.

I would probably not mention streams, to be honest. You don't use them much in Flutter apps at first.